test/suites/host-ip-change.robot

```diff
@@ -0,0 +1,229 @@
+    ${hostip}    ${rc2}=    Execute Command    ip route show | grep -i ens3 | grep -i default | awk '{print $9}'    return_stdout=True    return_rc=True
+
+    Should Be Equal As Integers    ${rc1}    0
+    Should Be Equal As Integers    ${rc2}    0
+
+    FOR    ${i}    IN RANGE    10    # Maximum 10 attempts to find an unassigned IP
+        ${last_digit}    Evaluate    int($hostip.split('.')[-1])
+        ${random_digit}    Evaluate    random.randint(0, 9)
+        ${new_host_ip}    Evaluate    '${hostip}'.rsplit('.', 1)[0] + '.23' + str($random_digit)
+        Log    Attempt ${i}: Generated IP: ${new_host_ip}
+        ${ping_output}    Execute Command    ping -c 1 ${new_host_ip} | grep -i "100% packet loss"| wc -l
+        Run Keyword If    ${ping_output} > 0    Exit For Loop
+    END
+
+    Log    Original Host IP: ${hostip}
+    Log    New Host IP: ${new_host_ip}
+
+    # Add one more check to ensure the generated IP is not the same as the current IP
+    Should Not Be Equal    ${new_host_ip}    ${hostip}
+
+    [Return]    ${new_host_ip}   ${gateway}
+
+Save Backup Default File
+    [Arguments]    ${file_path}
+    [Documentation]    Takes a backup of the given file and returns the path to the backup file.
+    ...    This keyword is meant to be used from a Setup step.
+    ${stdout}    ${rc}=    Execute Command
+    ...    cp ${file_path} ${file_path}.backup
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    ${rc}    0
+
+Restore Backup File
+    [Arguments]    ${backup_file}
+    [Documentation]    Restores the original file from the backup file.
+    ${stdout}    ${rc}=    Execute Command
+    ...    cp ${backup_file}.backup ${backup_file}
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    ${rc}    0
+
+Setup Suite
+    [Documentation]    Test suite setup
+    Check Required Env Variables
+    Login MicroShift Host
+    Setup Kubeconfig    # for readiness checks
+
+Rollback Configuration
+    [Arguments]    ${backup_file}
+    [Documentation]    Restore the original configuration from the backup file
+    Restore Backup File    ${backup_file}
+
+Add Node to Hosts File
+    [Arguments]    ${ip}    ${node_name}
+    [Documentation]    Add the node name and IP address to the /etc/hosts file
+    ${stdout}    ${rc}=    Execute Command
+    ...    echo "${ip} ${node_name}" | sudo tee -a /etc/hosts
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    ${rc}    0
+
+Get NodeName
+    [Documentation]    Get and Return nodename
+    ${nodeName}=    Run With Kubeconfig    oc get node --no-headers | awk '{print $1}'
+    Should Not Be Empty    ${nodeName}
+    [Return]    ${nodeName}
+
+Modify NetworkManager Configuration
+    [Arguments]    ${ip}    ${gateway}
+    [Documentation]    Modify NetworkManager configuration using nmcli commands
+    ${stdout}   ${rc1}=    Execute Command    nmcli con modify ens3 ipv4.addresses ${ip}
```

Comment:
The device name should be deduced programmatically.

Reply:
Updated